parsing/parser.mly

```diff
@@ -1686,6 +1691,25 @@ simple_pattern_not_ident:
   | extension
       { mkpat(Ppat_extension $1) }
 ;
+simple_pattern_arl:
```

Comment:
It stood for `array_record_list`. A better name might be `simple_delimited_pattern` or maybe simply
`simple_pattern_array_record_list`.